Error when calling catkin_make (with potential fix)

Ubuntu 16.04
ROS Kinetic

I'm getting the following errors when running `catkin_make`:

```
/home/administrator/catkin_ws/src/ros_kortex/kortex_examples/src/full_arm/example_cartesian_poses_with_notifications.cpp:36:18: error: variable 'std::atomic<int> last_action_notification_event' has initializer but incomplete type
 std::atomic<int> last_action_notification_event{0};
                  ^
/home/administrator/catkin_ws/src/ros_kortex/kortex_examples/src/full_arm/example_cartesian_poses_with_notifications.cpp:37:18: error: variable 'std::atomic<int> last_action_notification_id' has initializer but incomplete type
 std::atomic<int> last_action_notification_id{0};
                  ^
make[2]: *** [ros_kortex/kortex_examples/CMakeFiles/example_cartesian_poses_with_notifications_cpp.dir/src/full_arm/example_cartesian_poses_with_notifications.cpp.o] Error 1
make[1]: *** [ros_kortex/kortex_examples/CMakeFiles/example_cartesian_poses_with_notifications_cpp.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
In file included from /home/administrator/catkin_ws/src/ros_kortex/kortex_driver/src/non-generated/driver/kortex_arm_simulation.cpp:13:0:
/home/administrator/catkin_ws/src/ros_kortex/kortex_driver/include/kortex_driver/non-generated/kortex_arm_simulation.h:172:23: error: field 'm_is_action_being_executed' has incomplete type 'std::atomic<bool>'
     std::atomic<bool> m_is_action_being_executed;
                       ^
In file included from /usr/include/c++/5/bits/shared_ptr_atomic.h:33:0,
                 from /usr/include/c++/5/memory:83,
                 from /usr/include/boost/config/no_tr1/memory.hpp:21,
                 from /usr/include/boost/smart_ptr/shared_ptr.hpp:23,
                 from /usr/include/boost/shared_ptr.hpp:17,
                 from /usr/include/boost/format/alt_sstream.hpp:21,
                 from /usr/include/boost/format/internals.hpp:23,
                 from /usr/include/boost/format.hpp:38,
                 from /usr/include/boost/math/policies/error_handling.hpp:31,
                 from /usr/include/boost/math/special_functions/round.hpp:14,
                 from /opt/ros/kinetic/include/ros/time.h:58,
                 from /opt/ros/kinetic/include/ros/ros.h:38,
                 from /home/administrator/catkin_ws/src/ros_kortex/kortex_driver/include/kortex_driver/non-generated/kortex_arm_simulation.h:17,
                 from /home/administrator/catkin_ws/src/ros_kortex/kortex_driver/src/non-generated/driver/kortex_arm_simulation.cpp:13:
/usr/include/c++/5/bits/atomic_base.h:126:12: note: declaration of 'struct std::atomic<bool>'
     struct atomic;
            ^
In file included from /home/administrator/catkin_ws/src/ros_kortex/kortex_driver/src/non-generated/driver/kortex_arm_simulation.cpp:13:0:
/home/administrator/catkin_ws/src/ros_kortex/kortex_driver/include/kortex_driver/non-generated/kortex_arm_simulation.h:173:23: error: field 'm_action_preempted' has incomplete type 'std::atomic<bool>'
     std::atomic<bool> m_action_preempted;
                       ^
In file included from /usr/include/c++/5/bits/shared_ptr_atomic.h:33:0,
                 from /usr/include/c++/5/memory:83,
                 from /usr/include/boost/config/no_tr1/memory.hpp:21,
                 from /usr/include/boost/smart_ptr/shared_ptr.hpp:23,
                 from /usr/include/boost/shared_ptr.hpp:17,
                 from /usr/include/boost/format/alt_sstream.hpp:21,
                 from /usr/include/boost/format/internals.hpp:23,
                 from /usr/include/boost/format.hpp:38,
                 from /usr/include/boost/math/policies/error_handling.hpp:31,
                 from /usr/include/boost/math/special_functions/round.hpp:14,
                 from /opt/ros/kinetic/include/ros/time.h:58,
                 from /opt/ros/kinetic/include/ros/ros.h:38,
                 from /home/administrator/catkin_ws/src/ros_kortex/kortex_driver/include/kortex_driver/non-generated/kortex_arm_simulation.h:17,
                 from /home/administrator/catkin_ws/src/ros_kortex/kortex_driver/src/non-generated/driver/kortex_arm_simulation.cpp:13:
/usr/include/c++/5/bits/atomic_base.h:126:12: note: declaration of 'struct std::atomic<bool>'
     struct atomic;
            ^
In file included from /home/administrator/catkin_ws/src/ros_kortex/kortex_driver/src/non-generated/driver/kortex_arm_simulation.cpp:13:0:
/home/administrator/catkin_ws/src/ros_kortex/kortex_driver/include/kortex_driver/non-generated/kortex_arm_simulation.h:174:22: error: field 'm_current_action_type' has incomplete type 'std::atomic<int>'
     std::atomic<int> m_current_action_type;
                      ^
In file included from /usr/include/c++/5/bits/shared_ptr_atomic.h:33:0,
                 from /usr/include/c++/5/memory:83,
                 from /usr/include/boost/config/no_tr1/memory.hpp:21,
                 from /usr/include/boost/smart_ptr/shared_ptr.hpp:23,
                 from /usr/include/boost/shared_ptr.hpp:17,
                 from /usr/include/boost/format/alt_sstream.hpp:21,
                 from /usr/include/boost/format/internals.hpp:23,
                 from /usr/include/boost/format.hpp:38,
                 from /usr/include/boost/math/policies/error_handling.hpp:31,
                 from /usr/include/boost/math/special_functions/round.hpp:14,
                 from /opt/ros/kinetic/include/ros/time.h:58,
                 from /opt/ros/kinetic/include/ros/ros.h:38,
                 from /home/administrator/catkin_ws/src/ros_kortex/kortex_driver/include/kortex_driver/non-generated/kortex_arm_simulation.h:17,
                 from /home/administrator/catkin_ws/src/ros_kortex/kortex_driver/src/non-generated/driver/kortex_arm_simulation.cpp:13:
/usr/include/c++/5/bits/atomic_base.h:126:12: note: declaration of 'struct std::atomic<int>'
     struct atomic;
            ^
make[2]: *** [ros_kortex/kortex_driver/CMakeFiles/kortex_arm_driver_implementation.dir/src/non-generated/driver/kortex_arm_simulation.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [ros_kortex/kortex_driver/CMakeFiles/kortex_arm_driver_implementation.dir/all] Error 2
make: *** [all] Error 2
Invoking "make -j4 -l4" failed
```

I've also attached the rest of the output of `catkin_make` as well in case you need it. 

I added `#include <atomic>` to the following files fixed the problem:
- `ros_kortex/kortex_examples/src/full_arm/example_cartesian_poses_with_notifications.cpp`
- `ros_kortex/kortex_driver/include/kortex_driver/non-generated/kortex_arm_simulation.h`

Not sure if that is where these includes should go but I figured you would know that better than me!

[ros_kortex_catkin_make.txt](https://github.com/Kinovarobotics/ros_kortex/files/5037940/ros_kortex_catkin_make.txt)

--- a/kortex_description/launch/visualize.launch
+++ b/kortex_description/launch/visualize.launch
@@ -14,10 +14,10 @@
 
     <!-- Load the description for the robot -->
     <!-- Without gripper -->
-    <param name="robot_description" command="$(find xacro)/xacro $(find kortex_description)/robots/$(arg arm).xacro dof:=$(arg dof) vision:=$(arg vision) sim:=true"
+    <param name="robot_description" command="$(find xacro)/xacro --inorder $(find kortex_description)/robots/$(arg arm).xacro dof:=$(arg dof) vision:=$(arg vision) sim:=true"
         if="$(eval not arg('gripper'))"/>
     <!-- With gripper -->
-    <param name="robot_description" command="$(find xacro)/xacro $(find kortex_description)/robots/$(arg arm)_$(arg gripper).xacro dof:=$(arg dof) vision:=$(arg vision) sim:=true"
+    <param name="robot_description" command="$(find xacro)/xacro --inorder $(find kortex_description)/robots/$(arg arm)_$(arg gripper).xacro dof:=$(arg dof) vision:=$(arg vision) sim:=true"
         unless="$(eval not arg('gripper'))"/>
 
     <!-- Joint state publisher and robot state publisher -->

--- a/kortex_description/readme.md
+++ b/kortex_description/readme.md
@@ -23,10 +23,10 @@ To load the description of a robot, you simply have to load the **ARM.xacro** or
 For example:
 
 - To load the Gen3 description with a Robotiq 2-F 85 gripper for simulation, you would put in your launch file : 
-<code><param name="robot_description" command="$(find xacro)/xacro $(find kortex_description)/robots/gen3_robotiq_2f_85.xacro sim:=true"\/></code>
+<code><param name="robot_description" command="$(find xacro)/xacro --inorder $(find kortex_description)/robots/gen3_robotiq_2f_85.xacro sim:=true"\/></code>
 
 - To load the Gen3 lite description, you would put in your launch file : 
-<code><param name="robot_description" command="$(find xacro)/xacro $(find kortex_description)/robots/gen3_lite_gen3_lite_2f.xacro sim:=false"\/></code>
+<code><param name="robot_description" command="$(find xacro)/xacro --inorder $(find kortex_description)/robots/gen3_lite_gen3_lite_2f.xacro sim:=false"\/></code>
 
 ## Tool frame
 

--- a/kortex_driver/include/kortex_driver/non-generated/kortex_arm_simulation.h
+++ b/kortex_driver/include/kortex_driver/non-generated/kortex_arm_simulation.h
@@ -35,6 +35,7 @@
 #include <unordered_map>
 #include <thread>
 #include <mutex>
+#include <atomic>
 
 // Kortex
 #include "kortex_driver/non-generated/kortex_math_util.h"

--- a/kortex_driver/launch/kortex_driver.launch
+++ b/kortex_driver/launch/kortex_driver.launch
@@ -38,19 +38,19 @@
         <group if="$(eval not arg('prefix'))">
             <!-- Load the description for the robot -->
             <!-- Without gripper -->
-            <param name="robot_description" command="$(find xacro)/xacro $(find kortex_description)/robots/$(arg arm).xacro dof:=$(arg dof) vision:=$(arg vision) sim:=false"
+            <param name="robot_description" command="$(find xacro)/xacro --inorder $(find kortex_description)/robots/$(arg arm).xacro dof:=$(arg dof) vision:=$(arg vision) sim:=false"
                 if="$(eval not arg('gripper'))"/>
             <!-- With gripper -->
-            <param name="robot_description" command="$(find xacro)/xacro $(find kortex_description)/robots/$(arg arm)_$(arg gripper).xacro dof:=$(arg dof) vision:=$(arg vision) sim:=false"
+            <param name="robot_description" command="$(find xacro)/xacro --inorder $(find kortex_description)/robots/$(arg arm)_$(arg gripper).xacro dof:=$(arg dof) vision:=$(arg vision) sim:=false"
                 unless="$(eval not arg('gripper'))"/>
         </group>
         <group unless="$(eval not arg('prefix'))">
             <!-- Load the description for the robot -->
             <!-- Without gripper -->
-            <param name="robot_description" command="$(find xacro)/xacro $(find kortex_description)/robots/$(arg arm).xacro dof:=$(arg dof) vision:=$(arg vision) sim:=false prefix:=$(arg prefix)"
+            <param name="robot_description" command="$(find xacro)/xacro --inorder $(find kortex_description)/robots/$(arg arm).xacro dof:=$(arg dof) vision:=$(arg vision) sim:=false prefix:=$(arg prefix)"
                 if="$(eval not arg('gripper'))"/>
             <!-- With gripper -->
-            <param name="robot_description" command="$(find xacro)/xacro $(find kortex_description)/robots/$(arg arm)_$(arg gripper).xacro dof:=$(arg dof) vision:=$(arg vision) sim:=false prefix:=$(arg prefix)"
+            <param name="robot_description" command="$(find xacro)/xacro --inorder $(find kortex_description)/robots/$(arg arm)_$(arg gripper).xacro dof:=$(arg dof) vision:=$(arg vision) sim:=false prefix:=$(arg prefix)"
                 unless="$(eval not arg('gripper'))"/>
         </group>
 

--- a/kortex_driver/launch/kortex_dual_driver.launch
+++ b/kortex_driver/launch/kortex_dual_driver.launch
@@ -60,7 +60,7 @@
     <!-- Load arms descriptions -->
    <param 
         name="robot_description" 
-        command="$(find xacro)/xacro $(find kortex_description)/multiple_robots/kortex_dual_robots.xacro 
+        command="$(find xacro)/xacro --inorder $(find kortex_description)/multiple_robots/kortex_dual_robots.xacro 
             left_arm:=$(arg left_arm) 
             left_gripper:=$(arg left_gripper)
             left_dof:=$(arg left_dof)

--- a/kortex_examples/src/full_arm/example_cartesian_poses_with_notifications.cpp
+++ b/kortex_examples/src/full_arm/example_cartesian_poses_with_notifications.cpp
@@ -10,6 +10,7 @@
  *
  */
 #include <thread>
+#include <atomic>
 
 #include "ros/ros.h"
 #include <kortex_driver/Base_ClearFaults.h>

--- a/kortex_gazebo/launch/spawn_kortex_robot.launch
+++ b/kortex_gazebo/launch/spawn_kortex_robot.launch
@@ -45,10 +45,10 @@
 
         <!-- Load the description for the robot -->
         <!-- Without gripper -->
-        <param name="robot_description" command="$(find xacro)/xacro $(find kortex_description)/robots/$(arg arm).xacro dof:=$(arg dof) vision:=$(arg vision) sim:=true prefix:=$(arg prefix)"
+        <param name="robot_description" command="$(find xacro)/xacro --inorder $(find kortex_description)/robots/$(arg arm).xacro dof:=$(arg dof) vision:=$(arg vision) sim:=true prefix:=$(arg prefix)"
             if="$(eval not arg('gripper'))"/>
         <!-- With gripper -->
-        <param name="robot_description" command="$(find xacro)/xacro $(find kortex_description)/robots/$(arg arm)_$(arg gripper).xacro dof:=$(arg dof) vision:=$(arg vision) sim:=true prefix:=$(arg prefix)"
+        <param name="robot_description" command="$(find xacro)/xacro --inorder $(find kortex_description)/robots/$(arg arm)_$(arg gripper).xacro dof:=$(arg dof) vision:=$(arg vision) sim:=true prefix:=$(arg prefix)"
             unless="$(eval not arg('gripper'))"/>
 
         <!-- Spawn the robot in Gazebo -->

--- a/kortex_move_it_config/gen3_lite_gen3_lite_2f_move_it_config/launch/planning_context.launch
+++ b/kortex_move_it_config/gen3_lite_gen3_lite_2f_move_it_config/launch/planning_context.launch
@@ -13,10 +13,10 @@
 
   <!-- The semantic description that corresponds to the URDF -->
   <!-- With prefix -->
-  <param name="$(arg robot_description)_semantic" command="$(find xacro)/xacro $(find gen3_lite_gen3_lite_2f_move_it_config)/config/gen3_lite_gen3_lite_2f.srdf.xacro prefix:=$(arg prefix)"
+  <param name="$(arg robot_description)_semantic" command="$(find xacro)/xacro --inorder $(find gen3_lite_gen3_lite_2f_move_it_config)/config/gen3_lite_gen3_lite_2f.srdf.xacro prefix:=$(arg prefix)"
     unless="$(eval not arg('prefix'))"/>
   <!-- No prefix -->
-  <param name="$(arg robot_description)_semantic" command="$(find xacro)/xacro $(find gen3_lite_gen3_lite_2f_move_it_config)/config/gen3_lite_gen3_lite_2f.srdf.xacro"
+  <param name="$(arg robot_description)_semantic" command="$(find xacro)/xacro --inorder $(find gen3_lite_gen3_lite_2f_move_it_config)/config/gen3_lite_gen3_lite_2f.srdf.xacro"
     if="$(eval not arg('prefix'))"/>
   <!-- Load updated joint limits (override information from URDF) -->
   <group ns="$(arg robot_description)_planning">

--- a/kortex_move_it_config/gen3_move_it_config/launch/planning_context.launch
+++ b/kortex_move_it_config/gen3_move_it_config/launch/planning_context.launch
@@ -19,10 +19,10 @@
 
   <!-- The semantic description that corresponds to the URDF -->
   <!-- With prefix -->
-  <param name="$(arg robot_description)_semantic" command="$(find xacro)/xacro $(find gen3_move_it_config)/config/$(arg dof)dof/gen3.srdf.xacro prefix:=$(arg prefix)"
+  <param name="$(arg robot_description)_semantic" command="$(find xacro)/xacro --inorder $(find gen3_move_it_config)/config/$(arg dof)dof/gen3.srdf.xacro prefix:=$(arg prefix)"
     unless="$(eval not arg('prefix'))"/>
   <!-- No prefix -->
-  <param name="$(arg robot_description)_semantic" command="$(find xacro)/xacro $(find gen3_move_it_config)/config/$(arg dof)dof/gen3.srdf.xacro"
+  <param name="$(arg robot_description)_semantic" command="$(find xacro)/xacro --inorder $(find gen3_move_it_config)/config/$(arg dof)dof/gen3.srdf.xacro"
     if="$(eval not arg('prefix'))"/>
 
   <!-- Load updated joint limits (override information from URDF) -->

--- a/kortex_move_it_config/gen3_robotiq_2f_140_move_it_config/launch/planning_context.launch
+++ b/kortex_move_it_config/gen3_robotiq_2f_140_move_it_config/launch/planning_context.launch
@@ -19,10 +19,10 @@
 
   <!-- The semantic description that corresponds to the URDF -->
   <!-- With prefix -->
-  <param name="$(arg robot_description)_semantic" command="$(find xacro)/xacro $(find gen3_robotiq_2f_140_move_it_config)/config/$(arg dof)dof/gen3_robotiq_2f_140.srdf.xacro prefix:=$(arg prefix)"
+  <param name="$(arg robot_description)_semantic" command="$(find xacro)/xacro --inorder $(find gen3_robotiq_2f_140_move_it_config)/config/$(arg dof)dof/gen3_robotiq_2f_140.srdf.xacro prefix:=$(arg prefix)"
     unless="$(eval not arg('prefix'))"/>
   <!-- No prefix -->
-  <param name="$(arg robot_description)_semantic" command="$(find xacro)/xacro $(find gen3_robotiq_2f_140_move_it_config)/config/$(arg dof)dof/gen3_robotiq_2f_140.srdf.xacro"
+  <param name="$(arg robot_description)_semantic" command="$(find xacro)/xacro --inorder $(find gen3_robotiq_2f_140_move_it_config)/config/$(arg dof)dof/gen3_robotiq_2f_140.srdf.xacro"
     if="$(eval not arg('prefix'))"/>
 
   <!-- Load updated joint limits (override information from URDF) -->

--- a/kortex_move_it_config/gen3_robotiq_2f_85_move_it_config/launch/planning_context.launch
+++ b/kortex_move_it_config/gen3_robotiq_2f_85_move_it_config/launch/planning_context.launch
@@ -19,10 +19,10 @@
 
   <!-- The semantic description that corresponds to the URDF -->
   <!-- With prefix -->
-  <param name="$(arg robot_description)_semantic" command="$(find xacro)/xacro $(find gen3_robotiq_2f_85_move_it_config)/config/$(arg dof)dof/gen3_robotiq_2f_85.srdf.xacro  prefix:=$(arg prefix)"
+  <param name="$(arg robot_description)_semantic" command="$(find xacro)/xacro --inorder $(find gen3_robotiq_2f_85_move_it_config)/config/$(arg dof)dof/gen3_robotiq_2f_85.srdf.xacro  prefix:=$(arg prefix)"
     unless="$(eval not arg('prefix'))"/>
   <!-- No prefix -->
-  <param name="$(arg robot_description)_semantic" command="$(find xacro)/xacro $(find gen3_robotiq_2f_85_move_it_config)/config/$(arg dof)dof/gen3_robotiq_2f_85.srdf.xacro"
+  <param name="$(arg robot_description)_semantic" command="$(find xacro)/xacro --inorder $(find gen3_robotiq_2f_85_move_it_config)/config/$(arg dof)dof/gen3_robotiq_2f_85.srdf.xacro"
     if="$(eval not arg('prefix'))"/>
 
   <!-- Load updated joint limits (override information from URDF) -->

